File: ReSharper.FSharp/src/FSharp.Psi.Daemon/src/Analyzers/ZoneMarker.fs
namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Daemon.Analyzers

open JetBrains.Application.BuildScript.Application.Zones
open JetBrains.ReSharper.Feature.Services.Daemon
open JetBrains.ReSharper.Features.ReSpeller
open JetBrains.ReSharper.Plugins.FSharp

[<ZoneMarker>]
type ZoneMarker() =
    interface IRequire<DaemonEngineZone>
    interface IRequire<DaemonZone>
    interface IRequire<ILanguageFSharpZone>
    interface IRequire<IReSpellerZone>
    interface IRequire<ISinceClr4HostZone>

File: ReSharper.FSharp/src/FSharp/FSharp.Psi/src/ZoneMarker.cs
using JetBrains.Application.BuildScript.Application.Zones;

namespace JetBrains.ReSharper.Plugins.FSharp.Psi
{
  [ZoneMarker]
  public class ZoneMarker : IRequire<IFSharpPluginZone>;
}

namespace JetBrains.ReSharper.Plugins.FSharp.Services
{
  [ZoneMarker]
  public class ZoneMarker : IRequire<IFSharpPluginZone>;
}
